salt/_modules/metalk8s_kubernetes_cronjob.py

```diff
@@ -0,0 +1,212 @@
+        ]
+
+    return results
+
+
+def get_cronjob(name, namespace, **kwargs):
+    """Get a specific CronJob by name and namespace.
+
+    Args:
+        name (str): name of the CronJob.
+        namespace (str): namespace of the CronJob.
+
+    Raises:
+        CommandExecutionError: if the CronJob is not found.
+
+    Returns:
+        dict: the CronJob object.
+    """
+
+    cronjob = __salt__["metalk8s_kubernetes.get_object"](
+        kind="CronJob",
+        apiVersion="batch/v1",
+        name=name,
+        namespace=namespace,
+        **kwargs,
+    )
+    if not cronjob:
+        raise CommandExecutionError(
+            f"CronJob {name} not found in namespace {namespace}"
+        )
+
+    return cronjob
+
+
+def _set_cronjob_suspend(name, namespace, suspend, **kwargs):
+    """Set the suspend state of a CronJob.
+
+    Args:
+        name (str): name of the CronJob.
+        namespace (str): namespace of the CronJob.
+        suspend (bool): the new suspend state.
+
+    Returns:
+        True: if the suspend state has been updated.
+    """
+    cronjob = get_cronjob(name, namespace, **kwargs)
+
+    if cronjob["spec"]["suspend"] != suspend:
+        __salt__["metalk8s_kubernetes.update_object"](
+            kind="CronJob",
+            apiVersion="batch/v1",
+            name=name,
+            namespace=namespace,
+            patch={"spec": {"suspend": suspend}},
+            old_object=cronjob,
+            **kwargs,
+        )
+
+    return True
+
+
+def suspend_cronjob(name, namespace, **kwargs):
+    """Suspend a CronJob.
+
+    Args:
+        name (str): name of the CronJob.
+        namespace (str): namespace of the CronJob.
+
+    Returns:
+        True: if the suspend state has been updated.
+    """
+    return _set_cronjob_suspend(name, namespace, suspend=True, **kwargs)
+
+
+def activate_cronjob(name, namespace, **kwargs):
+    """Activate a CronJob.
+
+    Args:
+        name (str): name of the CronJob.
+        namespace (str): namespace of the CronJob.
+
+    Returns:
+        True: if the suspend state has been updated.
+    """
+    return _set_cronjob_suspend(name, namespace, suspend=False, **kwargs)
+
+
+def get_jobs(cronjob_name, namespace, **kwargs):
+    """Get the Jobs created by a CronJob.
+
+    Args:
+        cronjob_name (str): name of the CronJob.
+        namespace (str): namespace of the CronJob.
+
+    Returns:
+        list: list of Jobs created by the CronJob.
+    """
+    cronjob_uid = get_cronjob(cronjob_name, namespace, **kwargs)["metadata"]["uid"]
+
+    # Get all Jobs in the namespace
+    jobs = __salt__["metalk8s_kubernetes.list_objects"](
+        kind="Job", apiVersion="batch/v1", namespace=namespace, **kwargs
+    )
+
+    if not jobs:
+        return []
+
+    # Filter Jobs created by the CronJob
+    filtered_jobs = []
+
+    for job in jobs:
+        owner_refs = job["metadata"].get("ownerReferences", [])
+        for owner_ref in owner_refs:
```

Comment:
I didn't checked the content of `ownerReferences` when it's empty if it's an empty list of `None` but just to be safe I would prefer to handle both (also I don't feel this `owner_refs` variable is very useful)

```suggestion
        for owner_ref in  job["metadata"].get("ownerReferences") or []:
```